db/write_batch.cc

```diff
@@ -2114,6 +2149,76 @@ class MemTableInserter : public WriteBatch::Handler {
     return s;
   }
 
+  Status MarkCommitWithTimestamp(const Slice& name,
```

Comment:
I think it would be nice to reuse/share some code between `MarkCommit` and `MarkCommitWithTimestamp`.

Reply:
I am thinking of keeping this mid-sized PR simple such that it does not change existing logic used intensively by production workloads at this sensitive time of year :).
I am also not very sure if future changes to timestamp/transaction will make these two methods' implementation diverge for more complex transaction write-policies, e.g. write-prepared/write-unprepared. Maybe it's not the worst idea just to leave them separate for now.